### PR TITLE
Add a probation period for propfinds to ensure they make request of server

### PR DESCRIPTION
**Pull request by jerryblakley**

### CODE REVIEW

**Comment by davidstrauss** [comment]
+1

**Comment by jerryblakley** [comment]
@davidstrauss If you get a chance to re-review, this is what I plan to merge to master on Thursday.

**Comment by davidstrauss** [comment]
This PR is considerably more extensive than the name suggests.

File: src/statcache.c [hunk 1]
```diff
@@ -511,8 +511,22 @@ time_t stat_cache_read_updated_children(stat_cache_t *cache, const char *path, G
...
+    static const char *funcname = "stat_cache_negative_entry";
+    // Allow propfinds in first 5 seconds of creation of negative entry,
+    // since modules often check for (non-)existence of file before creating it
+    const time_t probation_delay = 5;
+    time_t curtime;
+    curtime = time(NULL);
+    if (value.st.st_ctime + probation_delay > curtime) {
+        log_print(LOG_INFO, SECTION_FUSEDAV_STAT, "%s: still within probation period %s; ctime: %lu; probation: %lu", 
+                funcname, path, value.st.st_ctime, curtime);
+        return curtime;
```
Comment (davidstrauss) [inline src/statcache.c]: I'd prefer something that isn't the current time. Would it be okay to return zero in order to indicate that the "next propfind" is ASAP?

**Comment by davidstrauss** [comment]
Aside from my one comment, this looks good.

File: src/statcache.c (hunk 1 above)
Comment (jerryblakley) [reply src/statcache.c]: 754         else {
   755             time_t current_time = time(NULL);
   756             time_t next_time;
   757
   758             next_time = stat_cache_next_propfind(*value, path);
   759
   760             // Our "time to next propfind" is still in the future, so no propfind ...
   761             if (next_time > current_time) {

I'd prefer to return the 'time to next propfind', which is in this case, immediately.

**Pull request merged**